src/zenml/integrations/aws/flavors/sagemaker_orchestrator_flavor.py

```diff
@@ -42,6 +45,13 @@ class SagemakerOrchestratorSettings(BaseSettings):
         max_runtime_in_seconds: The maximum runtime in seconds for the
             processing job.
         processor_tags: Tags to apply to the Processor assigned to the step.
+        keep_alive_period_in_seconds: The time in seconds after which the
+            provisioned instance will be terminated if not used. This is only
+            applicable for TrainingStep type and it is not possible to use
+            TrainingStep type if the `output_data_s3_uri` is set to Dict[str, str].
+        use_training_steps_where_possible: Whether to use the TrainingStep
```

Comment:
As far as I can see, this parameter can be used both on the pipeline-level and on the step-level. While the naming makes sense on the pipeline level, on the step-level it is a bit unintuitive IMO. How about we rename it to `use_training_step`?

I realize that some steps might not be conducted as a training step and they have to be conducted as a `ProcessingStep` because of this condition: `elif isinstance(step_settings.output_data_s3_uri, dict):`. We can check and correct this behaviour through a pydantic validator in the step settings.

Reply:
Thanks,  I made it via validator, but want to clarify something. @schustmi also pulling you in.

A user does the following:
1. Set `use_training_step` to True on a step and to False on a pipeline => outcome? I assume 'True'
2. Set `use_training_step` to False on a step and to True on a pipeline => outcome? I assume 'False'
3. Nothing set in step => use pipeline
4. Nothing set in pipeline => use step

Speaking in code
```python
use_training_step_in_step = step_settings.model_dump(exclude_unset=True).get("use_training_step",None)
if use_training_step_in_step is not None:
    use_train_step = use_training_step_in_step
else:
    use_train_step = self.config.use_training_step
```

I implemented it like this for now, LMK is this is lame.

Reply:
The settings get applied like this:
- If a value is set on the pipeline, it gets applied to all steps.
- If a value is set for a specific step, it overwrites the generic value from the pipeline.

So 1-3 of your points are definitely correct, the fourth one I think as well. The step one is always used first, only if not it will fallback to the pipeline one.

One thing to keep in mind here: Your `exclude_unset` filter here will probably not work. This is because we write the full settings (including fields with unset/default values) to our DB. The reason for this is that we can then display it in the dashboard, and it might also prevent issues if default values were to change between different releases.
This however means that `use_training_step` will always be `True` in case not specified by the user, and it will never fall back to the config. To achieve that, you'd probably have to make it an `Optional[bool] = None` field use use the `None` value to check if it was set.

Reply:
```python
use_training_step = step_settings.use_training_step
if use_training_step is None:
  use_training_step = self.config.use_training_step
if use_training_step is None:
  use_training_step = True # Some default value if nothing is specified anywhere
```

Something like this